packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -324,3 +352,15 @@ function sanitizeTiming(name: string) {
   }
   return sanitized
 }
+
+function wrapOnNewLocation(onNewLocation: NewLocationListener): NewLocationListener {
```

Comment:
This is nice, and I think it may be re-used for other user-provided callbacks, like `onReady`. No need to do it now though.

Reply:
I think it could be valuable to do it in a following PR though, with `beforeSend` also

Reply:
Ok, I will do it in a separate PR after merging this one.